### Fisher - Lady Tasting Tea

The original problem was to determine the probability of the lady correctly picking the 4 cups from 8 with milk in first, if it was through purely random selection. The probability is approx. 1.5 %


This exercise is to calculate the minimum number of cups of tea required to ensure the probability of randomly selecting the correct cups is less than or equal to 1%.


The following code is adapted from - https://stackoverflow.com/a/4941846

In [1]:
# Importing Math Module

import math

If selecting 4 cups from 8 gives a 1.5% chance, then we need to increase the sample size to get a probability closer to 1%.

In [3]:
# Selecting 4 cups from 9 - number of possible combinations
math.comb(9,4)

126

In [5]:
# Probability of correctly selecting 4 from 9 -rounded to 2 places

round(1/126*100, 2)


0.79

From the above, we can see that increasing the sample size to 9 gives an approx. 0.79 % chance of randomly selecting the correct 4 cups. 

If we try 4 cups from 10, we can see the chances of selecting the correct cups further reduced.

In [9]:
round(1/math.comb(10,4)*100, 2)

0.48

Selecting 4 cups from 10 gives an approx. 0.48% chance of randomly choosing the correct cups.